adobe-private/react-spectrum-v3
# RSP-1793 [Accessibility][Docs] Fix Docs site accessibility issues

1. All page areas need to be in a landmark region. This one isn’t, and I recommend role=”banner”:
  <div class="_pageHeader_2dc44" id="header">
2. The Three H3 headings on the home page should be H2’s
  a. Same on react stately/aria/spectrum pages
3. The “learn more” links are ambiguous since they are all the same. Options:
  a. Change the text to include more context, e.g., “Learn more about React Spectrum”
  b. Add hidden text, e.g., <a class="_spectrum-Link_f6693" href="./react-spectrum/index.html">Learn more <span class=”hidevisuallybutnotfromSR”> about React Spectrum</span></a>
  c. Use aria-label, e.g., <a class="_spectrum-Link_f6693" href="./react-spectrum/index.html" aria-label=”Learn more about React Spectrum”>Learn more</a>
4. Color contrast for the code/properties is too low:
  a. Green, e.g., 'false'
  b. Orange, e.g., UNSAFE_className
  c. Yellow
  d. Gray // Code Comment text
  e. Fuschia
  f. Indigo
5. Table heading cells in tables should use th element. Yes, they are within a THEAD, but TH is the right element to use.
  a. <td class="_spectrum-Table-headCell_b8fb4">Name</td>


Closes https://jira.corp.adobe.com/browse/RSP-1793

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- Added/updated unit tests and storybook for this change (for new code or code which already has tests).

## 📝 Test Instructions:
1. `yarn start:docs`
2. Open Documentation site site http://localhost:1234/index.html
3. Use aXe extension to check accessibility on docs pages.
4. Following issues should be addressed:
- `div#header` now has `[role="banner"]` on all pages, particularly in responsive mode
- Headings for "React Spectrum", "React Aria", and "React Stately" are now `H2`s
- "Learn more" links now use `VisuallyHidden` component to hide text that distinguishes between the links for screen reader users. e.g. "Learn more about React Aria"
- Color contrast for highlighted code examples should now be sufficient to meet WCAG 2.1 minimum luminosity contrast ratio of 4.5:1.
- Back button is now labeled with `alt` text on the ChevronLeft icon element
- Table headers for tables explaining element types now have `role="columnheader"` for columns and `role="rowheader"` for rows.
- Table of contents `nav` element, for in-page navigation, is now labelled to distinguish it from the main SideNav for the site.
- Adobe logo is now labelled, which distinguishes home page link from React Spectrum package link in SideNav.
- FocusRing and useFocusRing examples have been updated so that the foreground text on the example component has sufficient contrast.
- useListBox and useMenu examples have been updated to label the standalone listbox and menu examples.
- useSeparator example has been updated so that code example does not require horizontal scrolling.

TODO: Still requiring attention is the keyboard focus styling for the SideNav and links.

## 🧢 Your Team:

Accessibility
